src/snowflake/snowpark/table.py

```diff
@@ -314,26 +335,55 @@ def __init__(
         self.is_cached: bool = self.is_cached  #: Whether the table is cached.
         self.table_name: str = table_name  #: The table name
         self._is_temp_table_for_cleanup = is_temp_table_for_cleanup
+        self._time_travel_config = time_travel_config
 
         # By default, the set the initial API call to say 'Table.__init__' since
         # people could instantiate a table directly. This value is overwritten when
         # created from Session object
         set_api_call_source(self, "Table.__init__")
 
     def _copy_without_ast(self):
+        kwargs = {}
+        if self._time_travel_config:
+            kwargs.update(
+                {
+                    "time_travel_mode": self._time_travel_config.mode,
+                    "statement": self._time_travel_config.statement,
+                    "offset": self._time_travel_config.offset,
+                    "timestamp": self._time_travel_config.timestamp,
+                    "timezone": self._time_travel_config.timezone,
+                    "stream": self._time_travel_config.stream,
+                }
+            )
+
         return Table(
             self.table_name,
             session=self._session,
             is_temp_table_for_cleanup=self._is_temp_table_for_cleanup,
             _emit_ast=False,
+            **kwargs,
```

Comment:
I think you can try using `_asdict()` method on a namedtuple which should simplify your code
https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict